### PR TITLE
add drag coefficient and roughness to cache

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Cache drag coefficient and roughness as 2D fields. This is needed by the coupler.

### CODE REVIEW

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl [hunk 1]
```diff
@@ -303,9 +319,10 @@ function saturated_surface_conditions(
     params,
     coupled,
 )
-    Cd = surface_scheme.Cd
-    Ch = surface_scheme.Ch
-    # parameters
+    Cd = parent(p.Cd[colidx])[1]
```
Comment (szy21) [inline examples/hybrid/sphere/baroclinic_wave_utilities.jl:322]: If I don't do this I got `Cannot convert an object of type Field{} (trunc disp) to an object of type Float32`

Comment (LenkaNovak) [reply examples/hybrid/sphere/baroclinic_wave_utilities.jl:322]: Hmm would it be better to pass in Cd and Ch as en explicit argument to `saturated_surface_conditions`? Maybe in the long term, we can separate the cache into fields and other objects, so we can just pass in `p.fields_vector[colidx]`, but that change may break many things, so probably not worth it right now.

**Comment by szy21** [comment]
Oh this is allocating a lot...https://buildkite.com/clima/climaatmos-ci/builds/4408#0183d822-6472-4081-9183-e01830927310/140-186 @charleskawczynski is there anything I did wrong in the cache?

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl (hunk 1 above)
Comment (charleskawczynski) [inline examples/hybrid/sphere/baroclinic_wave_utilities.jl:322]:
```suggestion
    Cd = Spaces.level(p.Cd[colidx], 1)
```

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl [hunk 2]
```diff
@@ -303,9 +319,10 @@ function saturated_surface_conditions(
...
+    Ch = parent(p.Ch[colidx])[1]
```
Comment (charleskawczynski) [inline examples/hybrid/sphere/baroclinic_wave_utilities.jl:323]:
```suggestion
    Ch = Spaces.level(p.Ch[colidx], 1)
```

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl [hunk 3]
```diff
@@ -353,8 +372,9 @@ function saturated_surface_conditions(
     params,
     coupled,
 )
-    z0m = surface_scheme.z0m
-    z0b = surface_scheme.z0b
+    z0m = parent(p.z0m[colidx])[1]
+    z0b = parent(p.z0b[colidx])[1]
```
Comment (charleskawczynski) [inline examples/hybrid/sphere/baroclinic_wave_utilities.jl:376]:
```suggestion
    z0m = Spaces.level(p.z0m[colidx], 1)
    z0b = Spaces.level(p.z0b[colidx], 1)
```

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl (hunk 1 above)
Comment (charleskawczynski) [reply examples/hybrid/sphere/baroclinic_wave_utilities.jl:322]: I think we shouldn't ever need parent anymore

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**